hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -138,6 +148,18 @@ public void start() {
         proxyService.init();
         operationService.start();
         quorumService.start();
+        performanceMonitor.start();
+
+        // static loggers at beginning of file
+        performanceMonitor.register(new BuildInfoPlugin(this));
```

Comment:
THe NodeEngineImpl is acting like a application context; so the application gets constructed here. That is why it is perfectly fine that the NodeEngineImpl is violating this class data abstraction coupling.